x/programs/rust/examples/multisig/src/lib.rs

```diff
@@ -0,0 +1,495 @@
+}
+
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub struct Voter {
+    address: Address,
+    voted: bool,
+}
+
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub enum ProposalError {
+    AlreadyExecuted,
+    AlreadyVoted,
+    InexistentProposal,
+    QuorumNotReached,
+    NotVoter,
+    ExecutionFailed(ExternalCallError),
+}
+
+#[state_keys]
+pub enum StateKeys {
+    LastProposalId,
+    Proposals(u64),
+    Voter(u64, usize),
+    Yeas(u64),
+    Nays(u64),
+}
+
+#[public]
+pub fn propose(
+    context: Context<StateKeys>,
+    voters_addr: Vec<Address>,
+    to: Program,
+    method: String,
+    args: Vec<u8>,
+    value: u64,
+) -> u64 {
+    let voters_len = voters_addr.len().try_into().expect("too much voters");
+    assert!(voters_len >= MIN_VOTES);
+    assert!(_is_not_duplicate(&voters_addr));
+    let mut voters: Vec<_> = voters_addr
+        .iter()
+        .map(|address| Voter {
+            address: *address,
+            voted: false,
+        })
+        .collect();
+    let i = _ensure_voter(&context.actor(), &voters).unwrap();
+    voters[i as usize].voted = true;
+
+    let last_id = _last_proposal_id(&context);
+
+    context
+        .store_by_key(
+            StateKeys::Proposals(last_id),
+            &Proposal {
+                to,
+                method,
+                args,
+                value,
+                executed: false,
+                voters_len,
+            },
+        )
+        .unwrap();
+
+    for (i, voter) in voters.into_iter().enumerate() {
+        context
+            .store_by_key(StateKeys::Voter(last_id, i), &voter)
+            .unwrap();
+    }
+
+    context
+        .store([
+            (StateKeys::Nays(last_id), &0u64),
+            (StateKeys::Yeas(last_id), &1),
+        ])
+        .unwrap();
+
+    context
+        .store_by_key(StateKeys::LastProposalId, &(last_id + 1))
+        .unwrap();
+
+    last_id
+}
+
+#[public]
+pub fn vote(context: Context<StateKeys>, id: u64, yea: bool) -> Result<(), ProposalError> {
+    let proposal = _proposal_at(&context, id).ok_or(ProposalError::InexistentProposal)?;
+
+    if proposal.executed {
+        return Err(ProposalError::AlreadyExecuted);
+    }
+
+    let actor = context.actor();
+    let voters: Vec<_> = (0..proposal.voters_len)
+        .map(|i| {
+            context
+                .get(StateKeys::Voter(id, i as usize))
+                .unwrap()
+                .unwrap()
+        })
+        .collect();
+
+    let i = _ensure_voter(&actor, &voters)?;
+
+    if yea {
+        context
+            .store_by_key(
+                StateKeys::Yeas(id),
+                &(context.get::<u64>(StateKeys::Yeas(id)).unwrap().unwrap() + 1),
+            )
+            .unwrap();
+    } else {
+        context
+            .store_by_key(
+                StateKeys::Nays(id),
+                &(context.get::<u64>(StateKeys::Nays(id)).unwrap().unwrap() + 1),
+            )
+            .unwrap();
+    }
```

Comment:
This isn't very readable. Please store the result from `get` in a variable and use that. 

Please don't use `unwrap` either

Reply:
Thanks, created a new variable and provided some better panic messages with `expect`.